src/lib/components/ContentfulRichText/predicates.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  BLOCKS,
+  INLINES,
+  type Node,
+  type Document,
+  type Paragraph,
+  type Text,
+  type Hr,
+  type UnorderedList,
+  type OrderedList,
+  type ListItem,
+  type Hyperlink,
+  type Table,
+  type TableRow,
+  type TableHeaderCell,
+  type TableCell,
+  type Quote,
+} from "@contentful/rich-text-types";
+
+import type { HeadingLevel, HeadingTypeByLevel } from "./headings";
+
+export const isNode = (n: unknown): n is Node =>
+  typeof n === "object" &&
+  n !== null &&
+  "nodeType" in n &&
+  typeof n.nodeType === "string" &&
+  "data" in n &&
+  typeof n.data === "object";
+
+export const isDocument = (n: unknown): n is Document =>
+  isNode(n) && n.nodeType === BLOCKS.DOCUMENT;
+
+export const isParagraph = (n: unknown): n is Paragraph =>
+  isNode(n) && n.nodeType === BLOCKS.PARAGRAPH;
+
+export const isHeading = <L extends HeadingLevel>(
+  level: L,
+  n: unknown
+): n is HeadingTypeByLevel<L> => isNode(n) && n.nodeType === `heading-${level}`;
+
+export const isHr = (n: unknown): n is Hr => isNode(n) && n.nodeType === BLOCKS.HR;
+
+export const isText = (n: unknown): n is Text => isNode(n) && n.nodeType === "text";
+
+export const isQuote = (n: unknown): n is Quote => isNode(n) && n.nodeType === BLOCKS.QUOTE;
+
+export const isUnorderedList = (n: unknown): n is UnorderedList =>
+  isNode(n) && n.nodeType === BLOCKS.UL_LIST;
+
+export const isOrderedList = (n: unknown): n is OrderedList =>
+  isNode(n) && n.nodeType === BLOCKS.OL_LIST;
+
+export const isListItem = (n: unknown): n is ListItem =>
+  isNode(n) && n.nodeType === BLOCKS.LIST_ITEM;
+
+export const isHyperlink = (n: unknown): n is Hyperlink =>
+  isNode(n) && n.nodeType === INLINES.HYPERLINK;
+
+export const isTable = (n: unknown): n is Table => isNode(n) && n.nodeType === BLOCKS.TABLE;
+export const isTableRow = (n: unknown): n is TableRow =>
+  isNode(n) && n.nodeType === BLOCKS.TABLE_ROW;
+export const isTableHeaderCell = (n: unknown): n is TableHeaderCell =>
+  isNode(n) && n.nodeType === BLOCKS.TABLE_HEADER_CELL;
```

Comment:
Any idea why this is coming up as uncovered in the coverage report? The other uncovered lines seem like other common issues we've seen with the reporter (exporting and assigning values mostly), but we definitely have table headers in the test content; is markdown not rendering them to HTML appropriately?

This is super unimportant, especially since I think you're already doing a great job of covering all our bases here; don't sweat spending more than a couple minutes looking at this!

Reply:
It's because `@contentful/rich-text-from-markdown`, which is where all our test cases come from, is buggy and only renders table rows and cells, not table header rows and cells, even though the latter are defined in `@contentful/rich-text-types`. You can confirm this by running `node src/lib/components/ContentfulRichText/__tests__/documents/.generate-documents-from-markdown.cjs` yourself and confirming that the generated documents have no table header cells in JSON nor HTML.

Reply:
See https://github.com/contentful/rich-text/issues/352 for this issue.

Reply:
Good to know it's not an issue on our side! I would usually recommend leaving a comment to remind us why we're seeing this, but I would assume that this will eventually be fixed with a dependency bump so I'm cool with leaving everything as-is. Thanks for looking into this!